### PR TITLE
Add iommu groups and drivers info to resource pool

### PR DESCRIPTION
**Pull request by lvfxx**

### CODE REVIEW

File: pkg/sriov/types.go [hunk 1]
```diff
@@ -196,10 +230,34 @@ func (n *NetResourcePool) GetFreeVirtualFunctionsInfo() *FreeVirtualFunctionsInf
...
+	// TODO think about what we do with already configured devices
+	if n.sriovProvider.IsSriovConfigured(ctx, pciAddr) {
+		return errors.Errorf("device %s is already configured", pciAddr)
+	}
+
+	return nil
+}
+
 // PhysicalFunction contains information about physical function
 type PhysicalFunction struct {
```
Comment (Bolodya1997) [inline pkg/sriov/types.go:255]: It looks like `PhysicalFunction` duplicates all `VirtualFunction` fields, so please consider using extension:
```
type PhysicalFunction struct {
    TargetPCIAddress         string
    Capability               string
    VirtualFunctionsCapacity int
    VirtualFunctions         map[*VirtualFunction]VirtualFunctionState

    VirtualFunction
}
```

File: pkg/sriov/types.go [hunk 2]
```diff
@@ -32,96 +32,130 @@ type VirtualFunctionState string
...
+	// VfioPCIDriver is vfio-pci driver type
+	VfioPCIDriver DriverType = "vfio-pci"
 )
 
 // NetResourcePool provides contains information about net devices
 type NetResourcePool struct {
 	HostName          string
 	PhysicalFunctions []*PhysicalFunction
+	IommuGroups       map[int]DriverType
+	sriovProvider     utils.SriovProvider
```
Comment (Bolodya1997) [inline pkg/sriov/types.go:57]: Do we need `utils.SriovProvider` as a part of `NetResourcePool`? I believe it should be only used in `InitResourcePool` and so should be passed in there as an argument.

File: pkg/sriov/types.go [hunk 3]
```diff
@@ -32,96 +32,130 @@ type VirtualFunctionState string
...
 }
 
 // InitResourcePool configures devices, specified in provided config and initializes resource pool with that devices
 func InitResourcePool(ctx context.Context, config *ResourceDomain) (*NetResourcePool, error) {
-	resourcePool := &NetResourcePool{
+	rp := &NetResourcePool{
 		HostName:          config.HostName,
 		PhysicalFunctions: nil,
+		IommuGroups:       map[int]DriverType{},
+		sriovProvider:     utils.NewSriovProvider(pciDevicesPath, pciDriversPath, iommuGroupsPath),
```
Comment (Bolodya1997) [inline pkg/sriov/types.go:67]: I believe such things make testing difficult. Let it be passed to the `InitResourcePool` as an argument?

File: pkg/sriov/types.go (hunk 1 above)
Comment (lvfxx) [reply pkg/sriov/types.go:255]: Good idea :)

Comment (lvfxx) [reply pkg/sriov/types.go:255]: Initialization example:
``` go
physfun := &PhysicalFunction{
			VirtualFunction: VirtualFunction{
				PCIAddress:       pfPciAddr,
				BoundDriver:      KernelDriver, // Kernel driver is bound by default
				KernelDriverName: pfKernelDriverName,
				IommuGroup:       pfIommuGroup,
				NetInterfaceName: pfIfaceNames[0],
			},
			TargetPCIAddress:         device.Target.PCIAddress,
			Capability:               device.Capability,
			VirtualFunctionsCapacity: vfCapacity,
			VirtualFunctions:         map[*VirtualFunction]VirtualFunctionState{},
		}
```

File: pkg/sriov/types.go (hunk 2 above)
Comment (lvfxx) [reply pkg/sriov/types.go:57]: I agree with you

File: pkg/sriov/types.go (hunk 1 above)
Comment (lvfxx) [reply pkg/sriov/types.go:255]: What about having something like that?
```go
type NetworkFunction struct {
	PCIAddress       string
	BoundDriver      DriverType
	KernelDriverName string
	IommuGroup       int
	NetInterfaceName string
}

type VirtualFunction NetworkFunction

type PhysicalFunction struct {
        NetworkFunction
        other fields...
}
```

**Pull request merged**